### PR TITLE
Improve user creation user settings

### PR DESCRIPTION
**Pull request by thecalcc**
CPCN-150

### CODE REVIEW

File: assets/company-admin/components/CompanyAdminApp.tsx
Comment (tomaskikutis) [inline assets/company-admin/components/CompanyAdminApp.tsx:223]: Don't default to an empty string. Maybe the idea is that when you're creating a new user, it is `null`. That would make sense to keep it then. If you convert it to a string there would no longer be a way to differentiate between a real ID and ID missing. Unless you check the length, but that's not a good approach.

File: assets/company-admin/components/CompanyAdminApp.tsx
Comment (tomaskikutis) [inline assets/company-admin/components/CompanyAdminApp.tsx:65]: state only has a few fields, if you're touching the component already it'd be nice to type state too

File: assets/users/components/EditUser.tsx
Comment (tomaskikutis) [inline assets/users/components/EditUser.tsx:317]: move info from propTypes to `IProps`

File: assets/company-admin/components/CompanyAdminApp.tsx
Comment (thecalcc) [reply assets/company-admin/components/CompanyAdminApp.tsx:216]: I have moved it to IProps

File: assets/users/components/EditUser.tsx
Comment (thecalcc) [reply assets/users/components/EditUser.tsx:317]: I have already moved that info to IProps

**Comment by thecalcc** [comment]
FIX CI & E2E before merging

File: e2e/package-lock.json
Comment (tomaskikutis) [inline e2e/package-lock.json:1]: add npm rc and disallow package lock in this dir - with package lock it would run old tests or you'd need to regenerate the lock file on every commit

https://github.com/superdesk/superdesk/blob/master/client/.npmrc

File: e2e/running-e2e-locally.md
Comment (tomaskikutis) [inline e2e/running-e2e-locally.md:3]: looks like npm install step is missing

**Review by tomaskikutis — APPROVED** [review]

**Review by tomaskikutis — APPROVED** [review]

**Comment by thecalcc** [comment]
Don't merge until ts-migration-v2 is checked by QA.

**Pull request closed without merging**
